hastexo/tasks.py

```diff
@@ -160,24 +160,34 @@ def run(self, **kwargs):
         self.providers = []
         for provider in stack.providers:
             p = Provider.init(provider["name"])
-            p.set_logger(logger)
-            p.set_capacity(provider["capacity"])
+            if p:
+                p.set_logger(logger)
+                p.set_capacity(provider["capacity"])
 
-            template = read_from_contentstore(
-                stack.course_id,
-                provider["template"]
-            )
-            p.set_template(template)
-
-            environment_path = provider.get("environment")
-            if environment_path:
-                environment = read_from_contentstore(
+                template = read_from_contentstore(
                     stack.course_id,
-                    environment_path
+                    provider["template"]
                 )
-                p.set_environment(environment)
+                p.set_template(template)
 
-            self.providers.append(p)
+                environment_path = provider.get("environment")
+                if environment_path:
+                    environment = read_from_contentstore(
+                        stack.course_id,
+                        environment_path
+                    )
+                    p.set_environment(environment)
+
+                self.providers.append(p)
+            else:
+                logger.warning(
+                    f'Failed to initialize provider: {provider["name"]}, '
+                    f'make sure the necessary settings are in place or remove '
+                    f'the provider from the list to silence this warning.')
```

Comment:
Minor note: as far as I know, you can mix f-strings with normal strings in this case, like so:

```python
logger.warning(
    f'Failed to initialize provider: {provider["name"]}, '
    'make sure the necessary settings are in place or remove '
    'the provider from the list to silence this warning.')
```
I tend to prefer that because it tells me at a glance when a part of a string is meant to be substituted, and when it's not.

Also, equally minor: insert comma before "or".

Reply:
Good note, updated now, thanks!